vlm/src/main/scala/geotrellis/contrib/vlm/MosaicRasterSource.scala

```diff
@@ -99,7 +100,9 @@ trait MosaicRasterSource extends RasterSource {
   )
 
   def read(extent: Extent, bands: Seq[Int]): Option[Raster[MultibandTile]] = {
-    val rasters = sources map { _.read(extent, bands) }
+    val rasters = sources map {
+      _.read(extent, bands).map(_.crop(extent, Crop.Options(false, false)))
```

Comment:
We've talked with @echeipesh and @moradology, and discovered that there is a problem in the constructor: that MosaicRasterSource constructor resamples to a potentially incorrect grid in the incorrect CRS the original rasterSource: https://github.com/geotrellis/geotrellis-contrib/blob/master/vlm/src/main/scala/geotrellis/contrib/vlm/MosaicRasterSource.scala#L156

So this fix is not a fix and breaks all the logic.